script/upload

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -x
+
+export PATH=/snap/bin:$PATH
+
+charm whoami
+RET=$?
+if ((RET > 0)); then
+    echo "Not logged into charmstore"
+    exit 1
+fi
+
+rm -rf temp
+mkdir temp
+cd temp
+unzip ../"$CHARM".charm
+URL=$(charm push . cs:~"$NAMESPACE"/"$CHARM" | yq r - url)
+charm attach cs:~"$NAMESPACE"/"$CHARM" --channel unpublished sriov-cni-image=sriov-cni
+
+IMAGE_NAME=$(charm list-resources "$URL" --format yaml | yq r - [0].name)
+IMAGE_REV=$(charm list-resources "$URL" --format yaml | yq r - [0].revision)
```

Comment:
I'm worried that due to charm store de-duping, `$URL` could end up pointing to a charm revision that already exists in another channel. If that happens, then this will grab the resource revision from the "most stable" channel that the revision exists in, i.e. *not* the unpublished channel.

I recommend using this instead:
```
charm list-resources cs:~"$NAMESPACE"/"$CHARM" --channel unpublished
```